Mathlib/LinearAlgebra/Span/Defs.lean

```diff
@@ -349,6 +349,12 @@ theorem mem_sup : x ∈ p ⊔ p' ↔ ∃ y ∈ p, ∃ z ∈ p', y + z = x :=
 theorem mem_sup' : x ∈ p ⊔ p' ↔ ∃ (y : p) (z : p'), (y : M) + z = x :=
   mem_sup.trans <| by simp only [Subtype.exists, exists_prop]
 
+theorem codisjoint_iff_exists_add_eq :
+    Codisjoint p p' ↔ ∀ z, ∃ x y, x ∈ p ∧ y ∈ p' ∧ x + y = z := by
+  rw [codisjoint_iff, eq_top_iff']
+  exact forall_congr' (fun z => mem_sup.trans <| by simp)
+
+@[deprecated codisjoint_iff_exists_add_eq (since := "2025-07-05")]
 lemma exists_add_eq_of_codisjoint (h : Codisjoint p p') (x : M) :
     ∃ y ∈ p, ∃ z ∈ p', y + z = x := by
   suffices x ∈ p ⊔ p' by exact Submodule.mem_sup.mp this
```

Comment:
You should actually delete the old lemma since there is deprecation:
```suggestion
```

Reply:
Sorry, judging by the comments this PR is not as straightforward as I thought it was. I had understood that one was to leave in place a working version of the old theorem but mark it as deprecated so that it could be deleted at some point in the future. What am I missing?